lms/static/js/spec/student_profile/learner_profile_factory_spec.js

```diff
@@ -62,6 +62,33 @@ define(['backbone', 'jquery', 'underscore', 'common/js/spec_helpers/ajax_helpers
                 LearnerProfileHelpers.expectLimitedProfileSectionsAndFieldsToBeRendered(learnerProfileView);
             });
 
+            it("doesn't render the mode toggle if there are no badges", function() {
```

Comment:
Is this a requirement? I don't see mention of it in UX-2980. IMHO it's confusing to hide this sometimes, and it means users with no badges won't see the "Earn Your Next Accomplishment" call to action.